src/hotspot/share/gc/parallel/mutableNUMASpace.cpp

```diff
@@ -892,28 +886,26 @@ void MutableNUMASpace::LGRPSpace::accumulate_statistics(size_t page_size) {
   clear_space_stats();
   char *start = (char*)align_up(space()->bottom(), page_size);
   char* end = (char*)align_down(space()->end(), page_size);
-  if (start < end) {
-    for (char *p = start; p < end;) {
-      os::page_info info;
-      if (os::get_page_info(p, &info)) {
-        if (info.size > 0) {
-          if (info.size > (size_t)os::vm_page_size()) {
-            space_stats()->_large_pages++;
-          } else {
-            space_stats()->_small_pages++;
-          }
-          if (info.lgrp_id == lgrp_id()) {
-            space_stats()->_local_space += info.size;
-          } else {
-            space_stats()->_remote_space += info.size;
-          }
-          p += info.size;
-        } else {
-          p += os::vm_page_size();
+  for (char *p = start; p < end; ) {
+    static const size_t PagesPerIteration = 128;
+    const void* pages[PagesPerIteration];
+    int lgrp_ids[PagesPerIteration];
+
+    size_t npages = 0;
+    for (; npages < PagesPerIteration && p < end; p += os::vm_page_size())
+      pages[npages++] = p;
```

Comment:
This means each page is `vm_page_size()` aligned. Would it be problematic if large-page is in use?

Reply:
I don't think so: for HugeTLBFS it makes some redundant calls but in my testing always returns the node ID for the containing huge page even if the address given is not huge page aligned, and for THP you don't know whether a particular address is backed by a huge page so you have to step with the smallest granularity anyway. With large pages enabled the "small/large pages" count is misleading but I don't know any way to get this information on Linux except by parsing `/proc/self/smaps` which we probably don't want to do.

Reply:
I see; thanks for the clarification. Re the misleading "small/large pages" count, can it be dropped? Or, what useful info does this count convey towards identifying wrongly placed OS pages?

Reply:
I don't think it adds any useful information now that the large page count is always zero: it's really just repeating the space size from the line above. I've removed it in the latest commit.